src/Apps/Settings/Routes/EditProfile/Components/SettingsEditProfileImage/SettingsEditProfileImage.tsx

```diff
@@ -3,31 +3,119 @@ import { CollectorProfileHeaderAvatarFragmentContainer } from "Apps/CollectorPro
 import { useFormikContext } from "formik"
-import { ChangeEvent, useState } from "react"
+import {
+  ChangeEvent,
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useState,
+} from "react"
 import { createFragmentContainer, graphql } from "react-relay"
 import styled from "styled-components"
+import {
+  getProfileLocalImage,
+  LocalImage,
+  storeLocalProfileImage,
+} from "Utils/localImagesHelpers"
 import { SettingsEditProfileImage_me$data } from "__generated__/SettingsEditProfileImage_me.graphql"
 
 interface SettingsEditProfileImageProps {
   me: SettingsEditProfileImage_me$data
 }
 
-const SettingsEditProfileImage: React.FC<SettingsEditProfileImageProps> = ({
-  me,
-}) => {
-  const [localImage, setlocalImage] = useState<string>("")
+export interface SettingsEditProfileImageRef {
+  storeImageLocally: () => Promise<void>
+}
+
+const SettingsEditProfileImage = forwardRef<
+  SettingsEditProfileImageRef,
+  SettingsEditProfileImageProps
+>(({ me }, ref) => {
   const { setFieldValue } = useFormikContext<EditProfileFormModel>()
+  const [localImage, setLocalImage] = useState<LocalImage>()
+  const [localImageBase64, setLocalImageBase64] = useState<string | null>(null)
+  const [isLoading, setIsLoading] = useState(true)
+
+  const convertFileToBase64 = (file: File) => {
+    // convert file to base64
+    const reader = new FileReader()
+    reader.readAsDataURL(file)
+    reader.onloadend = function () {
+      const base64data = reader.result
+      setLocalImageBase64(base64data as string)
+    }
+  }
+
+  const storeImageLocally = async () => {
+    // Store the image in local storage
+    if (localImage) {
+      return void (await storeLocalProfileImage(localImage))
+    }
+  }
+
+  useImperativeHandle(
+    ref,
+    () => {
+      return {
+        storeImageLocally,
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [storeImageLocally, localImage]
+  )
+
+  useEffect(() => {
+    getProfileLocalImage()
+      .then(image => {
+        if (image) {
+          setLocalImage(image)
+        }
+      })
+      .finally(() => setIsLoading(false))
+  }, [])
 
   const handleChange = async (event: ChangeEvent<HTMLInputElement>) => {
     if (!event.target.files) return
 
     const newImage = event.target.files[0]
-    const imagePreview = URL.createObjectURL(newImage)
 
-    setlocalImage(imagePreview)
+    convertFileToBase64(newImage)
     setFieldValue("photo", newImage)
   }
 
+  const handleImageLoad = (
+    image: React.SyntheticEvent<HTMLImageElement, Event>
+  ) => {
+    const {
+      naturalHeight: height,
+      naturalWidth: width,
+      currentSrc,
+    } = image.target as any
+
+    if (currentSrc.startsWith("data:image")) {
+      // Save the image dimensions as well as local path to the localImages array
+      setLocalImage({ data: currentSrc, width, height })
+    }
+  }
+
+  const renderProfileImage = () => {
```

Comment:
@olerichter00 this is not a recommended pattern but @MrSltun and I spoke about how we want to avoid more prop dilling here and that it would be fine in this case to do this.

Reply:
Looks good to me 👍